### PR TITLE
backup: allow backup tolerate minor TiKV failure

### LINKED ISSUE #980: BR should tolerate a small amount of tikv crash
### Describe your feature request related problem:

In the case of 9 tikv nodes, when a tikv crashes and cannot get up again, br will not work and report the log
```
error: cluster tidb1373933076652000599/test-330031, wait pipe message failed, errMsg [2021/04/01 03:29:14.541 +00:00] [ERROR] [push.go:54] ["fail to connect store"] [StoreID=70467] [stack="github.com/pingcap/br/pkg/backup.(*pushDown).pushBackup\n\tgithub.com/pingcap/br@/pkg/backup/push.go:54\ngithub.com/pingcap/br/pkg/backup.(*Client).BackupRange\n\tgithub.com/pingcap/br@/pkg/backup/client.go:524\ngithub.com/pingcap/br/pkg/backup.(*Client).BackupRanges.func2.1\n\tgithub.com/pingcap/br@/pkg/backup/client.go:459\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]
[2021/04/01 03:29:14.541 +00:00] [ERROR] [push.go:54] ["fail to connect store"] [StoreID=5417812] `[stack="github.com/pingcap/br/pkg/backup.(*pushDown).pushBackup\n\tgithub.com/pingcap/br@/pkg/backup/push.go:54\ngithub.com/pingcap/br/pkg/backup.(*Client).BackupRange\n\tgithub.com/pingcap/br@/pkg/backup/client.go:524\ngithub.com/pingcap/br/pkg/backup.(*Client).BackupRanges.func2.1\n\tgithub.com/pingcap/br@/pkg/backup/client.go:459\ngithub.com/pingcap/br/pkg/utils.(*WorkerPool).ApplyOnErrorGroup.func1\n\tgithub.com/pingcap/br@/pkg/utils/worker.go:63\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57"]

```
### Describe the feature you'd like:

When there is a problem with a small amount of tikv, but the cluster can work normally, should br continue to work instead of failing directly?

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve? 
fixed #980 

### What is changed and how it works?
ignore all ‘failed to connect to store’ errors during backup.

### Check List 

Tests 

 - Integration test

### Release Note

 - BR now can tolerate minor TiKV disconnection.

### CODE REVIEW

**Comment by YuJuncen** [comment]
After this merged, I think we need to add `br_serv_outage` to slow case, or strip some cases...

image

File: tests/br_serv_outage/run.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    [ "$count" -eq 20000 ]
+}
+
+load
+
+hint_finegrained=$TEST_DIR/hint_finegrained
+hint_backup_start=$TEST_DIR/hint_backup_start
+hint_get_backup_client=$TEST_DIR/hint_get_backup_client
+
+trap 'export GOFAILPOINTS=""' EXIT
```
Comment (kennytm) [inline tests/br_serv_outage/run.sh:56]:
```suggestion
```

File: pkg/backup/push.go [hunk 2]
```diff
@@ -51,6 +53,11 @@ func (push *pushDown) pushBackup(
 
 	// Push down backup tasks to all tikv instances.
 	res := rtree.NewRangeTree()
+	failpoint.Inject("noop-backup", func(_ failpoint.Value) {
+		log.Warn("skipping normal backup, jump to fine-grained backup, meow :3", logutil.Key("start-key", req.StartKey), logutil.Key("end-key", req.EndKey))
```
Comment (kennytm) [inline pkg/backup/push.go:57]: 😼

File: pkg/backup/client.go [hunk 3]
```diff
@@ -914,8 +959,9 @@ backupLoop:
 					}
 					break
 				}
-				return errors.Annotatef(err, "failed to connect to store: %d with retry times:%d", storeID, retry)
+				return berrors.ErrFailedToConnect.Wrap(err).GenWithStack("failed to connect to store: %d with retry times:%d: %s", storeID, retry, err.Error())
```
Comment (kennytm) [inline pkg/backup/client.go:962]: is it really necessary to include the `err` in GenWithStack? it is already present in Wrap...

```suggestion
				return berrors.ErrFailedToConnect.Wrap(err).GenWithStack("failed to connect to store: %d with retry times:%d", storeID, retry)
```

**Review by kennytm — COMMENTED** [review]
Rest LGTM

File: pkg/backup/client.go (hunk 3 above)
Comment (YuJuncen) [reply pkg/backup/client.go:962]: https://github.com/pingcap/errors/issues/33 (This is only for debugging, and I will remove them then. 🥲)

**Pull request merged**